evap/staff/importers/json.py

```diff
@@ -0,0 +1,204 @@
+            semester=self.semester,
+            cms_id=data["gguid"],
+            defaults={"name_de": data["title"], "name_en": data["title_en"], "type": course_type},
+        )
+        course.degrees.set(degrees)
+        course.responsibles.set(responsibles)
+
+        self.course_map[data["gguid"]] = course
+
+        return course
+
+    def _import_evaluation(self, course: Course, data: ImportEvent) -> Evaluation:
+        course_end = datetime.strptime(data["appointments"][0]["end"], self.DATETIME_FORMAT)
+
+        if data["isexam"]:
+            # Set evaluation time frame of three days for exam evaluations:
+            evaluation_start_datetime = course_end.replace(hour=8, minute=0) + timedelta(days=1)
+            evaluation_end_date = (course_end + timedelta(days=3)).date()
+
+            name_de = "Klausur"
+            name_en = "Exam"
+        else:
+            # Set evaluation time frame of two weeks for normal evaluations:
+            # Start datetime is at 8:00 am on the monday in the week before the event ends
+            evaluation_start_datetime = course_end.replace(hour=8, minute=0) - timedelta(
+                weeks=1, days=course_end.weekday()
+            )
+            # End date is on the sunday in the week the event ends
+            evaluation_end_date = (course_end + timedelta(days=6 - course_end.weekday())).date()
+
+            name_de, name_en = "", ""
+
+        # If events are graded for any degree, wait for grade upload before publishing
+        wait_for_grade_upload_before_publishing = any(filter(lambda grade: grade["scale"], data["courses"]))
+
+        participants = self._get_user_profiles(data["students"])
+
+        evaluation, __ = Evaluation.objects.update_or_create(
+            course=course,
+            cms_id=data["gguid"],
+            defaults={
+                "name_de": name_de,
+                "name_en": name_en,
+                "vote_start_datetime": evaluation_start_datetime,
+                "vote_end_date": evaluation_end_date,
+                "wait_for_grade_upload_before_publishing": wait_for_grade_upload_before_publishing,
+            },
+        )
+        evaluation.participants.set(participants)
+
+        for lecturer in data["lecturers"]:
+            self._import_contribution(evaluation, lecturer)
+
+        return evaluation
+
+    def _import_contribution(self, evaluation: Evaluation, data: ImportRelated):
+        user_profile = self.user_profile_map[data["gguid"]]
+
+        contribution, __ = Contribution.objects.update_or_create(
+            evaluation=evaluation,
+            contributor=user_profile,
+        )
+        return contribution
+
+    def _import_events(self, data: list[ImportEvent]):
+        # Divide in two lists so corresponding courses are imported before their exams
+        normal_events = filter(lambda e: not e["isexam"], data)
+        exam_events = filter(lambda e: e["isexam"], data)
```

Comment:
I think our linting will ask you to convert these to list comprehensions (same for `any(filter(lambda grade: grade["scale"], data["courses"]))` above in line 150).

I think `event for event in data if not event["isexam"]` is a bit more natural to read.